types/versionlockstring.pp

```diff
@@ -0,0 +1 @@
+type Yum::VersionlockString = Pattern[/^([0-9\*]+):([0-9a-zA-Z\._\+%\{\}\*-]+)-([^-]+)-([0-9\*]+)\.(([0-9a-zZ-Z_\*]+)(?:\.(noarch|x86_64|i386|arm|ppc64|ppc64le|sparc64|ia64|alpha|ip|m68k|mips|mipsel|mk68k|mint|ppc|rs6000|s390|s390x|sh|sparc|xtensa|\*))?)$/]
```

Comment:
It would be nice if types were composable, so you could do Pattern[^#{Yum::Version}:#{Yum::Epoch}] and this was readable and maintainable. But it's unfortunately they're not.

Maybe a documentation blurb breaking the regex out into it's components?